### ISSUE TITLE
Implement neural network and training pipeline prototype

### ISSUE THREAD

**Issue by williamstark01** [yangtcai/Ensembl-Repeat-Identification#12]
Potentially using [PyTorch Lightning](https://github.com/PyTorchLightning/pytorch-lightning) if it helps.

**Comment by yangtcai** [comment]
I think the issue can be divided into three parts
- The DETR model
   This part will produce the prediction results, the input of the model is the subsequence, and the output is the boundaries and the classes of the repeat sequences, the prototype of the DETR model is the following
```python
class DETR(nn.Module):
     def forward(sample):
         """
          Parameters:
              -- sample: batched sequences, of shape [batch_size x seq_len x embedding_len ]
                   eg, when choosing one hot encoding, embedding_len will be 5, [A, T, C, G, N]
              -- pred_logits: the classification logits (including no-object) for all queries.
                                Shape= [batch_size x num_queries x (num_classes + 1)]
              -- pred_boundaries: The normalized boundaries coordinates for all queries, represented as
                               (center, width). These values are normalized in [0, 1].
         """
          pass
```
- Hungarian matcher
   As the DETR model only produce `num_queries` prediction results, the order of prediction sets and target set has still not been decided, so the purpose of the part is that find the minimal matching between the prediction set and the target set.
- The Loss function
   It includes the classification loss and the segment loss.

**Comment by williamstark01** [comment]
@yangtcai that sounds like a good idea, feel free to open new issues for these subtasks and add them to the project board.

**Comment by williamstark01** [comment]
Implemented in #24
